identity/src/main/java/com/android/identity/MobileSecurityObjectGenerator.java

```diff
@@ -0,0 +1,272 @@
+package com.android.identity;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import co.nstant.in.cbor.CborBuilder;
+import co.nstant.in.cbor.builder.ArrayBuilder;
+import co.nstant.in.cbor.builder.MapBuilder;
+import co.nstant.in.cbor.model.UnicodeString;
+import java.security.PublicKey;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class MobileSecurityObjectGenerator {
+
+    private final String mDigestAlgorithm;
+    private final String mDocType;
+
+    private final CborBuilder mValueDigestsBuilder = new CborBuilder();
+    private final MapBuilder<CborBuilder> mValueDigestsOuter = mValueDigestsBuilder.addMap();
+    private Boolean digestEmpty = true;
```

Comment:
No need to use the boxed type `Boolean`, should be sufficient to just use the primitive `boolean` type.